cli/cmd/metrics_test.go

```diff
@@ -116,7 +116,7 @@ spec:
 	// However, only pod-1 is owned by deploy-1 according to the owner references.
 	// Owner references should be considered authoritative to resolve ambiguity
 	// when deployments have overlapping seletors.
-	pods, err := getPodsFor(context.Background(), k8sClient, "ns", "deploy/deploy-1")
+	pods, err := k8s.GetPodsFor(context.Background(), k8sClient, "ns", "deploy/deploy-1")
```

Comment:
As this exclusively tests the `GetPodsFor`. Should we move it into the k8s tests.